### PR TITLE
Feat/port

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

Ports all the extensions Ardian developed on academy-1 repo here.

Sibling PR: https://github.com/valory-xyz/agent-academy-1/pull/93

## Fixes

-

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [x] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [x] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

-

### CODE REVIEW

File: packages/valory/connections/abci/connection.py [hunk 1]
```diff
@@ -489,7 +1124,7 @@ def init(self) -> None:
         cmd = self._build_init_command()
         subprocess.call(cmd)  # nosec
 
-    def start(self, start_monitoring: bool = False) -> None:
```
Comment (DavidMinarsch) [inline packages/valory/connections/abci/connection.py:492]: Revert changes here and below

File: tests/helpers/docker/ganache.py [hunk 2]
```diff
@@ -106,3 +106,21 @@ def wait(self, max_attempts: int = 15, sleep_rate: float = 1.0) -> bool:
...
+    BLOCK_NUMBER: int = 11290000
+
+    def _build_command(self) -> List[str]:
+        """Build command."""
+
+        cmd = [
+            "--wallet.deterministic=true",
+            f"--fork.network={self.NETWORK}",
+            f"--fork.blockNumber={self.BLOCK_NUMBER}",
+        ]
```
Comment (marcofavorito) [inline tests/helpers/docker/ganache.py:124]: wrt parent class, we do not set gas limit and account/balances:

https://github.com/valory-xyz/open-autonomy/blob/35a90b02d2b30371994796a0935bb2568ef4f83e/tests/helpers/docker/ganache.py#L73-L81

Probably the solution is to call `super()._build_command()` and then append the new options?

Comment (DavidMinarsch) [reply tests/helpers/docker/ganache.py:124]: @0xArdi agree with @marcofavorito here, any reason you see why we shouldn't do that?

**Review by DavidMinarsch — COMMENTED** [review]
Fails on same tests as main, so merging now

File: tests/helpers/docker/ganache.py (hunk 2 above)
Comment (0xArdi) [reply tests/helpers/docker/ganache.py:124]: `--wallet.accounts` takes precedence over `--wallet.deterministic=true`, but that's not a problem.

**Pull request merged**